tests/virtual_machines_test.go

```diff
@@ -41,44 +44,409 @@ var _ = Describe("Virtual Machines", func() {
 	Context("Check the client", func() {
 		AfterEach(func() {
 			vmList := &kubevirtv1.VirtualMachineList{}
-			err := testClient.VirtClient.List(context.TODO(),&client.ListOptions{},vmList)
+			err := testClient.VirtClient.List(context.TODO(), &client.ListOptions{}, vmList)
 			Expect(err).ToNot(HaveOccurred())
 
 			for _, vmObject := range vmList.Items {
-				err = testClient.VirtClient.Delete(context.TODO(),&vmObject)
+				err = testClient.VirtClient.Delete(context.TODO(), &vmObject)
 				Expect(err).ToNot(HaveOccurred())
 			}
 		})
 
-		It("should create a vm object and automatically assign a static mac address", func() {
-			vm := CreateVmObject(TestNamespace,false,[]kubevirtv1.Interface{ovsInterface},[]kubevirtv1.Network{ovsNetwork})
-			Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).To(BeEmpty())
+		Context("When the client wants to create a vm", func() {
+			It("should create a vm object and automatically assign a static MAC address", func() {
 
-			err := testClient.VirtClient.Create(context.TODO(),vm)
-			Expect(err).ToNot(HaveOccurred())
+				vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{ovsInterface}, []kubevirtv1.Network{ovsNetwork})
+				Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).To(BeEmpty())
```

Comment:
I don't think you need the `Expect` after the `CreateVmObject` because is a function that we control not need to check anything there